reconcile/test/conftest.py

```diff
@@ -60,16 +60,16 @@ def s3_state_builder() -> Callable[[Mapping], State]:
     }
     """
 
-    def builder(data: Mapping) -> State:
-        def get(key: str, *args: Any) -> dict:
+    def builder(data: Mapping[str, Any]) -> State:
+        def get(key: str, *args: Any) -> dict[str, Any]:
             try:
                 return data["get"][key]
             except KeyError:
                 if args:
                     return args[0]
                 raise
 
-        def __getitem__(self: Any, key: str) -> dict:  # noqa: N807
+        def __getitem__(self: Any, key: str) -> dict[str, Any]:  # noqa: N807
```

Comment:
Please remove this obsolete annotation:


```suggestion
        def __getitem__(self, key: str) -> dict[str, Any]:  # noqa: N807
```